Get job from api

gdetrez opened the pull request:
- Refactor the daemon class to set the scanning job dynamically
- Add an `APIGateway` class to communicate with SoM API
- Add a new scheduled job to update the scanning job every minute

### Testing

This also add a `--skip-slow` flag to the test suite that skips tests marked as slow and mark the timing tests as slow.

The original system test was assuming that the daemon would start
scanning by itself.  Now, we need set-up a job on the server and pass
the proper credentials to the daemon for it to get the scanning
instruction.  The rest is the same (wait for the scanning to finish and
check the times at which the images appeared...)

I decided to experiment a bit with testing in production environment
("staging" actually but the idea is the same): instead of running an
isolated copy of scannomatic to run the tests, I use the existing
staging environement.  There are a few advantages:
- simpler and faster test setup
- tests are done in real condidions (e.g. with password authentication)
- tests are run on a Pi, which can't run scannomatic easily

There are also a few potential problems:
- there's only one scanner, which means there can't be concurrent test
  runs. This will be fixed once we can "register" scanners in some way
- there's a small chance that the test fails because scanomatic is
  redeployed.  If this is a problem, we can tell jenkins to coordinate
  job correctly in order to avoid it.

Commit:
Refactor ScanCommand as a callable that takes a job as argument

Commit:
Add a method to ScanDaemon to set the scanning job

Commit:
Add an APIGateway with a method to get the scanner's job

Commit:
Add UpdateScanningJobCommand

Commit:
Add job to update scanning instruction to daemon

Commit:
Update main to add job update

local-minimum commented on scanomaticd/apigateway.py:
```diff
@@ -0,0 +1,39 @@
+
+    def get_scanner_job(self, scannerid):
+        response = requests.get(
+            self.apibase + '/scanners/sc4nn3r/job'
```
Why is `sca4nn3r` hardcoded into the URI?

gdetrez replied:
fixed

local-minimum commented on scanomaticd/apigateway.py:
```diff
@@ -0,0 +1,39 @@
+        end = start + duration
+        return ScanningJob(
+            id=data['identifier'],
+            interval=timedelta(seconds=data['interval']),
```
In the UI interval is in minutes. Don't know if something has been done to it to make it seconds.

gdetrez replied:
Yes, and I think you're the one who approved it :stuck_out_tongue_winking_eye:

local-minimum replied:
memory-
```
|   |  | ,
----------
||  |  | _

```

Commit:
Remove hard-coded scanner id

Commit:
Add scannerid, username and password as arguments to APIGateway and authentify request

Commit:
Update scan job start field name in apigateway

Commit:
Get scanner id, api root, username and password from env

Commit:
Fix system tests

The original system test was assuming that the daemon would start
scanning by itself.  Now, we need set-up a job on the server and pass
the proper credentials to the daemon for it to get the scanning
instruction.  The rest is the same (wait for the scanning to finish and
check the times at which the images appeared...)

I decided to experiment a bit with testing in production environment
("staging" actually but the idea is the same): instead of running an
isolated copy of scannomatic to run the tests, I use the existing
staging environement.  There are a few advantages:
- simpler and faster test setup
- tests are done in real condidions (e.g. with password authentication)
- tests are run on a Pi, which can't run scannomatic easily

There are also a few potential problems:
- there's only one scanner, which means there can't be concurrent test
  runs. This will be fixed once we can "register" scanners in some way
- there's a small chance that the test fails because scanomatic is
  redeployed.  If this is a problem, we can tell jenkins to coordinate
  job correctly in order to avoid it.

Commit:
# Fix scanomaticd system tests
# Write integration tests for scanomatic api: register scanner, create job, start job, see that job is scanner's current job
# Add auth to APIGateway
Increase sleep time to fix test flakiness

Commit:
Fix jenkinsfile

Commit:
Remove development APIROOT

Commit:
Add missing auth when starting job in system test

Commit:
s/jobId/identifier/

local-minimum commented on system-tests/test_scanomaticd.py:
```diff
@@ -1,39 +1,76 @@
+        },
+        auth=(USERNAME, PASSWORD),
+    )
+    print(response.json())
```
remove?

Commit:
Remove print
